qiskit/ignis/verification/tomography/fitters/cvx_fit.py

```diff
@@ -24,8 +24,29 @@
 # Check if CVXPY package is installed
 try:
     import cvxpy
+    _HAS_CVX = True
 except ImportError:
-    cvxpy = None
+    _HAS_CVX = False
+
+
+def _default_sdp_solver():
+    solvers = cvxpy.installed_solvers()
+    if 'CVXOPT' in solvers:
+        return 'CVXOPT'
+    if 'SCS' in solvers:
+        # Try example problem to see if BLAS is installed
+        try:
+            var = cvxpy.Variable((4, 4), PSD=True)
+            obj = cvxpy.Minimize(cvxpy.norm(var))
+            cvxpy.Problem(obj).solve(solver='SCS')
+            return 'SCS'
+        except cvxpy.error.SolverError:
+            return None
+    return None
+
+
+# Check if a suitable default solver is available
+_DEFAULT_SOLVER = _default_sdp_solver()
```

Comment:
I am ok with this method for detecting which solver to use, but I am worried about doing this at the module level and the the impact on import performance. I think if you run `import qiskit` with this applied we'll end up running the 4x4 problem in the default case because cvxopt won't be installed. Can we move the execution to runtime instead of at import? We can preserve it as global state so we only run it once, just want to avoid it on import